docs/sources/flow/reference/components/loki.process.md

```diff
@@ -937,6 +947,597 @@ stage {
+metric value by 1 for each log line that passed the filter. The `add` action
+converts the extracted value to a positive float and adds it to the metric.
+
+
+#### gauge block
+Defines a gauge metric whose value can go up or down.
+
+The following arguments are supported:
+
+Name            | Type       | Description | Default | Required
+--------------- | ---------- | ----------- | ------- | --------
+`name`          | `string`   | The metric name. | | yes
+`action`        | `string`   | The action to take. Valid actions are `inc` and `add`. | | yes
+`description`   | `string`   | The metric's description and help text. | `""` | no
+`source`        | `string`   | Key from the extracted data map to use for the metric. Defaults to the metric name. | `""` | no
+`prefix`        | `string`   | The prefix to the metric name. | `"loki_process_custom_"` | no
+`idle_duration` | `duration` | Maximum amount of time to wait until the metric is marked as 'stale' and removed. | `"5m"` | no
+`value`         | `string`   | If set, the metric only changes if `source` exactly matches the `value`. | `""` | no
+
+
+The valid `action` values are `inc`, `dec`, `set`, `add`, or `sub`.
+`inc` and `dec` increment and decrement the metric's value by 1 respectively.
+If `set`, `add, or `sub` is chosen, the extracted value must be convertible
+to a positive float and is set, added or subtracted to the metric's value.
+
+
+#### histogram block
+Defines a histogram metric whose values are recorded in predefined buckets.
+
+
+The following arguments are supported:
+
+Name            | Type          | Description | Default | Required
+--------------- | ------------- | ----------- | ------- | --------
+`name`          | `string`      | The metric name. | | yes
+`buckets`       | `list(float)` | The action to take. Valid actions are `set`, `inc`, `dec`,` add`, or `sub`. | | yes
+`description`   | `string`      | The metric's description and help text. | `""` | no
+`source`        | `string`      | Key from the extracted data map to use for the metric. Defaults to the metric name. | `""` | no
+`prefix`        | `string`      | The prefix to the metric name. | `"loki_process_custom_"` | no
+`idle_duration` | `duration`    | Maximum amount of time to wait until the metric is marked as 'stale' and removed. | `"5m"` | no
+`value`         | `string`      | If set, the metric only changes if `source` exactly matches the `value`. | `""` | no
+
+#### metrics behavior
+
+If `value` is not present, all incoming log entries match.
+
+Label values on created metrics can be dynamic, which can cause exported
+metrics to explode in cardinality, or go stale, for example when a stream stops
+receiving new logs. To prevent unbounded growth of the /metrics endpoint, any
+metrics which have not been updated within `idle_duration` are removed. The
+`idle_duration` must be greater or equal to `"1s"`, and it defaults to `"5m"`
```

Comment:
```suggestion
Label values on created metrics can be dynamic, which can cause exported
metrics to explode in cardinality or go stale, for example, when a stream stops
receiving new logs. To prevent unbounded growth of the `/metrics` endpoint, any
metrics which have not been updated within `idle_duration` are removed. The
`idle_duration` must be greater or equal to `"1s"`, and it defaults to `"5m"`.
```